--- src/3rdparty/webkit/Source/WebCore/features.pri	2012-11-23 16:10:02.000000000 +0600
+++ src/3rdparty/webkit/Source/WebCore/features.pri.new	2013-05-03 22:31:22.507710944 +0600
@@ -156,6 +159,9 @@
 
 !contains(DEFINES, ENABLE_TOUCH_EVENTS=.): DEFINES += ENABLE_TOUCH_EVENTS=1
 
+# Explicitly disable video support
+DEFINES += ENABLE_VIDEO=0
+
 # HTML5 Media Support
 # We require QtMultimedia
 !contains(DEFINES, ENABLE_VIDEO=.) {
