--- src/3rdparty/webkit/Source/WebCore/rendering/RenderInline.cpp	2013-06-07 11:16:57.000000000 +0600
+++ src/3rdparty/webkit/Source/WebCore/rendering/RenderInline.cpp.new	2013-08-31 23:51:17.065280932 +0600
@@ -98,6 +98,8 @@
             if (firstLineBox()->parent()) {
                 for (InlineFlowBox* box = firstLineBox(); box; box = box->nextLineBox())
                     box->remove();
+            } else if (parent())  
+                parent()->dirtyLinesFromChangedChild(this); 
             }
         }
     }
