--- webkit-qtwebkit-23/Tools/Scripts/webkitdirs.pm	2013-04-24 21:37:47.000000000 +0600
+++ webkit-qtwebkit-23/Tools/Scripts/webkitdirs.pm.new	2013-06-28 16:42:54.047178352 +0600
@@ -834,12 +834,12 @@
 {
     die "ERROR: qmake missing but required to build WebKit.\n" if not commandExists($qmakebin);
 
-    my $qmakepath = File::Spec->catfile(sourceDir(), "Tools", "qmake");
+    my $qmakepath = File::Spec->catfile(windowsSourceDir(), "Tools", "qmake");
     my $qmakecommand;
     if (isWindows()) {
         $qmakecommand = "(set QMAKEPATH=$qmakepath) && $qmakebin";
     } else {
-        $qmakecommand = "QMAKEPATH=$qmakepath $qmakebin";
+        $qmakecommand = "$qmakebin";
     }
 
     my $originalCwd = getcwd();
@@ -2220,7 +2220,7 @@
     my @buildArgs = ();
     my $qconfigs = "";
 
-    my $make = qtMakeCommand($qmakebin);
+    my $make = "nmake";
     my $makeargs = "";
     my $installHeaders;
     my $installLibs;
@@ -2246,12 +2246,12 @@
         $makeargs .= " -j" . numberOfCPUs();
     }
 
-    my $qmakepath = File::Spec->catfile(sourceDir(), "Tools", "qmake");
+    my $qmakepath = File::Spec->catfile(windowsSourceDir(), "Tools", "qmake");
     my $qmakecommand;
     if (isWindows()) {
         $qmakecommand = "(set QMAKEPATH=$qmakepath) && $qmakebin";
     } else {
-        $qmakecommand = "QMAKEPATH=$qmakepath $qmakebin";
+        $qmakecommand = "$qmakebin";
     }
 
     my $config = configuration();
@@ -2380,7 +2380,7 @@
     if (! -e $makefile) {
         push @buildArgs, "-after OVERRIDE_SUBDIRS=\"@{$projects}\"" if @{$projects};
 
-        push @buildArgs, File::Spec->catfile(sourceDir(), "WebKit.pro");
+        push @buildArgs, File::Spec->catfile(windowsSourceDir(), "WebKit.pro");
         my $command = "$qmakecommand @buildArgs";
         print "Calling '$command' in " . $dir . "\n\n";
         print "Installation headers directory: $installHeaders\n" if(defined($installHeaders));
