--- src/corelib/tools/qiterator.h	2013-06-07 11:16:52.000000000 +0600
+++ src/corelib/tools/qiterator.h.new	2013-09-09 18:30:26.328998894 +0600
@@ -47,7 +47,7 @@
 QT_BEGIN_HEADER
 
 #ifdef QT_NO_STL
-# include <ciso646> // No-op, indirectly include additional configuration headers.
+# include <new> // No-op, indirectly include additional configuration headers.
 # if defined(_LIBCPP_VERSION)
 // libc++ may declare these structs in an inline namespace. Forward-declare
 // these iterators in the same namespace so that we do not shadow the original
