--- src/gui/styles/qgtkstyle.cpp	2012-11-23 16:09:53.000000000 +0600
+++ src/gui/styles/qgtkstyle.cpp.new	2012-12-20 06:50:52.000000000 +0600
@@ -1708,8 +1708,10 @@
 
 
             GtkRange *range = (GtkRange*)(horizontal ? gtkHScrollBar : gtkVScrollBar);
-            GtkAdjustment *adjustment = d->gtk_range_get_adjustment(range);
+            GtkAdjustment *adjustment = 0;
 
+            if (d->gtk_adjustment_configure)
+                adjustment = d->gtk_range_get_adjustment(range);
             if (adjustment) {
                 d->gtk_adjustment_configure(adjustment, fakePos, 0, maximum, 0, 0, 0);
             } else {
