diff --git a/Source/WebCore/platform/network/qt/PublicSuffixQt.cpp b/Source/WebCore/platform/network/qt/PublicSuffixQt.cpp
--- a/Source/WebCore/platform/network/qt/PublicSuffixQt.cpp
+++ b/Source/WebCore/platform/network/qt/PublicSuffixQt.cpp
@@ -64,7 +64,7 @@ String topPrivatelyControlledDomain(const String& domain)
         return domain;
 
     String lowercaseDomain = domain.convertToASCIILowercase();
-    if (lowercaseDomain == QLatin1String("localhost"))
+    if (lowercaseDomain == "localhost"_s)
         return lowercaseDomain;
 
     QString qLowercaseDomain = lowercaseDomain;
