diff --git a/Source/WebKit/PlatformQt.cmake b/Source/WebKit/PlatformQt.cmake
--- a/Source/WebKit/PlatformQt.cmake
+++ b/Source/WebKit/PlatformQt.cmake
@@ -106,25 +106,6 @@
 
     UIProcess/API/cpp/qt/WKStringQt.cpp
     UIProcess/API/cpp/qt/WKURLQt.cpp
-
-    UIProcess/API/qt/qquicknetworkreply.cpp
-    UIProcess/API/qt/qquicknetworkrequest.cpp
-    UIProcess/API/qt/qquickurlschemedelegate.cpp
-    UIProcess/API/qt/qquickwebpage.cpp
-    UIProcess/API/qt/qquickwebview.cpp
-    UIProcess/API/qt/qtwebsecurityorigin.cpp
-    UIProcess/API/qt/qwebchannelwebkittransport.cpp
-    UIProcess/API/qt/qwebdownloaditem.cpp
-    UIProcess/API/qt/qwebdownloaditem_p.h
-    UIProcess/API/qt/qwebdownloaditem_p_p.h
-    UIProcess/API/qt/qwebiconimageprovider.cpp
-    UIProcess/API/qt/qwebkittest.cpp
-    UIProcess/API/qt/qwebloadrequest.cpp
-    UIProcess/API/qt/qwebnavigationhistory.cpp
-    UIProcess/API/qt/qwebnavigationrequest.cpp
-    UIProcess/API/qt/qwebpermissionrequest.cpp
-    UIProcess/API/qt/qwebpreferences.cpp
-
     UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.cpp
 
     UIProcess/Launcher/qt/ProcessLauncherQt.cpp
@@ -192,19 +173,6 @@
     WebProcess/qt/WebProcessQt.cpp
 )
 
-if (COMPILER_IS_GCC_OR_CLANG)
-    set_source_files_properties(
-        UIProcess/API/qt/qquicknetworkreply.cpp
-        UIProcess/API/qt/qquicknetworkrequest.cpp
-        UIProcess/API/qt/qquickurlschemedelegate.cpp
-        UIProcess/API/qt/qquickwebpage.cpp
-        UIProcess/API/qt/qquickwebview.cpp
-        UIProcess/API/qt/qwebiconimageprovider.cpp
-    PROPERTIES
-        COMPILE_FLAGS -frtti
-    )
-endif ()
-
 if (USE_MACH_PORTS)
     list(APPEND WebKit_INCLUDE_DIRECTORIES
         "${WEBKIT_DIR}/Platform/IPC/cocoa"
@@ -251,15 +219,11 @@
 list(APPEND WebKit_SYSTEM_INCLUDE_DIRECTORIES
     ${GLIB_INCLUDE_DIRS}
     ${GSTREAMER_INCLUDE_DIRS}
-    ${Qt5Quick_INCLUDE_DIRS}
-    ${Qt5Quick_PRIVATE_INCLUDE_DIRS}
     ${SQLITE_INCLUDE_DIR}
 )
 
 list(APPEND WebKit_LIBRARIES
     ${Qt5Positioning_LIBRARIES}
-    ${Qt5Quick_LIBRARIES}
-    ${Qt5WebChannel_LIBRARIES}
     ${X11_X11_LIB}
 )
 
