diff --git a/qtbase/mkspecs/features/qt_parts.prf b/qtbase/mkspecs/features/qt_parts.prf
index ad87d46..3f6b36d 100644
--- a/qtbase/mkspecs/features/qt_parts.prf
+++ b/qtbase/mkspecs/features/qt_parts.prf
@@ -27,6 +27,7 @@ exists($$_PRO_FILE_PWD_/src/src.pro) {
     exists($$_PRO_FILE_PWD_/tools/tools.pro) {
         sub_tools.subdir = tools
         sub_tools.target = sub-tools
         sub_tools.depends = sub_src
+        !contains(QT_BUILD_PARTS, tools): sub_tools.CONFIG = no_default_target no_default_install
         # conditional treatment happens on a case-by-case basis
         SUBDIRS += sub_tools
     }
